src/engine/DnCManager.cpp

```diff
@@ -138,8 +141,11 @@ void DnCManager::solve( unsigned timeoutInSeconds )
     std::list<std::thread> threads;
     for ( unsigned threadId = 0; threadId < _numWorkers; ++threadId )
     {
-        threads.push_back( std::thread( dncSolve, workload,
-                                        _engines[ threadId ],
+        InputQuery *inputQuery = new InputQuery();
```

Comment:
who is going to deallocate the memory?

Reply:
Thanks for pointing this out. I wrap the InputQuery with a unique ptr. Since there will be a copy of input query for each worker, it still makes sense to allocate them on the heap.